server/sonar-server/src/main/java/org/sonar/server/computation/debt/Characteristic.java

```diff
@@ -44,6 +47,11 @@ public String getKey() {
     return key;
   }
 
+  @CheckForNull
+  public Integer getParentId() {
+    return parentId;
+  }
+
   @Override
   public boolean equals(@Nullable Object o) {
```

Comment:
is really a characteristic with a specific key not the same as another characteristic with the same key but a different id?